### PR TITLE
[util] added auto-doc

### PR DESCRIPTION
**Pull request by FrankLeeeee**
The user can embed something like `{{ autodoc:colossalai.amp.apex_amp.convert_to_apex_amp }}` in his markdown files and this will be converted to React code for its docstring in MDX.

Screenshot 2023-02-27 at 16 49 21

### CODE REVIEW

**Review by binmakeswell — APPROVED** [review]

File: doc-build/docer/cli/commands.py [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    auto_doc = AutoDoc()
+    
+    doc_dir = os.path.join(cache, 'docs')
+
+    # list all versions in this directory
+    for directory in os.listdir(doc_dir):
+        version_dir = os.path.join(doc_dir, directory)
+
+        # prepare page info
+        print(directory)
```
Comment (ver217) [inline doc-build/docer/cli/commands.py:41]: Is this necessary?

File: doc-build/setup.py [hunk 2]
```diff
@@ -15,7 +15,7 @@ def get_version():
         return f.read().strip()
 
 setup(
-    name='doc_build',
+    name='docer',
```
Comment (ver217) [inline doc-build/setup.py:18]: Should the package name be the same as folder name?

File: doc-build/docer/cli/commands.py (hunk 1 above)
Comment (FrankLeeeee) [reply doc-build/docer/cli/commands.py:41]: I will remove it.

File: doc-build/setup.py (hunk 2 above)
Comment (FrankLeeeee) [reply doc-build/setup.py:18]: it may confuse others as `doc_build` is similar to HF's `doc_builder`.

File: doc-build/docer/cli/commands.py (hunk 1 above)
Comment (FrankLeeeee) [reply doc-build/docer/cli/commands.py]: Done.

**Pull request merged**